module StatementAfterReturn

// Statement after exhaustive return errors
fun test1(): int {
  return 1;
  2;
}

fun test2(): int {
  return 1;
  return 2;
}

fun test3(): int {
  if (true) {
    return 1;
  } else {
    return 2;
  }

  3;
  return 4;
}

// Statement after inexhaustive return does not error
fun test4(): int {
  {
    1;
  }

  return 2;
}

fun test5(): int {
  if (true) {
    return 1;
  }

  2;
  return 3;
}

fun test6(): int {
  if (true) {
    1;
  } else {
    return 2;
  }

  3;
  return 4;
}
